controllers/discovery/initDiscoveryUserData.js

```diff
@@ -69,11 +88,21 @@ const InitDiscoveryUserData = async (req, res) => {
     });
     let suggestedUsers = usersWithCommonFollowerResult;
 
+    let totalData = await sequelize.query(totalDataQuery, {
+      type: QueryTypes.SELECT,
+      replacements: {
+        userId
+      },
+      raw: true
+    });
+    totalData = totalData[0]?.total_data;
+
     return res.status(200).json({
       success: true,
       message: `Fetch discovery data success`,
       suggestedUsers,
-      nextPage: page + 1
+      nextPage: page + 1,
+      total_page: Math.ceil(totalData / limit)
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The catch block only logs 'e' to the console, which doesn't provide any useful information about the error that occurred. It would be better to log the actual error object and also send a response with an appropriate status code and message.

```diff
- console.log('e');
+ console.error(e);
+ return res.status(500).json({
+   success: false,
+   message: 'An error occurred while fetching discovery data',
+ });
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->